lib/checkersreport.cpp

```diff
@@ -280,3 +280,89 @@ std::string CheckersReport::getReport(const std::string& criticalErrors) const
 
     return fout.str();
 }
+
+std::string CheckersReport::getXmlReport(const std::string& criticalErrors) const
+{
+    std::string ret;
+
+    ret += "    <critical-errors>";
+    if (!criticalErrors.empty())
+        ret += criticalErrors + "\n";
+    ret += "</critical-errors>\n";
+    ret += "    <checkers-report>\n";
+
+    const bool cppcheckPremium = isCppcheckPremium(mSettings);
+
+    auto reportSection = [&ret, cppcheckPremium]
+                             (const std::string& title,
+                             const Settings& settings,
+                             const std::set<std::string>& activeCheckers,
+                             const std::map<std::string, std::string>& premiumCheckers,
+                             const std::string& substring) {
+        ret += "        <" + title + ">";
+        if (!cppcheckPremium) {
+            ret += "Not available, Cppcheck Premium is not used</" + title + ">\n";
```

Comment:
this info is not targeted for machines but rather humans. I think it's better to just remove it.